src/test/projectsCreate.spec.ts

```diff
@@ -0,0 +1,220 @@
+import { expect } from "chai";
+import * as sinon from "sinon";
+
+import * as api from "../api";
+import { createFirebaseProject, ParentResource, ParentResourceType } from "../projectsCreate";
+import * as pollUtils from "../operation-poller";
+import { mockAuth } from "./helpers";
+import { OraWrapper } from "../oraWrapper";
+
+const PROJECT_ID = "the-best-firebase-project";
+const PROJECT_NUMBER = "1234567890";
+const PROJECT_NAME = "The Best Project";
+const PARENT_RESOURCE: ParentResource = { id: "1111111111111", type: ParentResourceType.FOLDER };
+const OPERATION_RESOURCE_NAME_1 = "operations/cp.11111111111111111";
+const OPERATION_RESOURCE_NAME_2 = "operations/cp.22222222222222222";
+
+// TODO(caot): Removed when "Deferred Analytics" and "Deferred Location" are launched
+const TIME_ZONE = "America/Los_Angeles";
+const REGION_CODE = "US";
+const LOCATION_ID = "us-central";
+
+describe("FirebaseResourceManager", () => {
+  describe("createFirebaseProject", () => {
+    let sandbox: sinon.SinonSandbox;
+    let mockOraWrapper: sinon.SinonMock;
+    let apiRequestStub: sinon.SinonStub;
+    let pollOperationStub: sinon.SinonStub;
+    let expectedCalledStubs: sinon.SinonStub[];
+
+    beforeEach(() => {
+      sandbox = sinon.createSandbox();
+      mockAuth(sandbox);
+      mockOraWrapper = sandbox.mock(OraWrapper.prototype);
+      apiRequestStub = sandbox.stub(api, "request");
+      pollOperationStub = sandbox.stub(pollUtils, "pollOperation");
+      expectedCalledStubs = [];
+    });
+
+    afterEach(() => {
+      mockOraWrapper.verify();
+      expectedCalledStubs.forEach((stub) => expect(stub.calledOnce).to.be.true);
+      sandbox.restore();
+    });
+
+    it("should resolve with project data if it succeeds", async () => {
```

Comment:
These tests are testing _two_ things: (1) that the correct calls are being made, and (2) that the spinner is working correctly (as evidenced by the mock and its assertions). I'm not convinced this is correct as it makes these tests rather extensive, and each test is describing what the function does and never mentions the spinner which is super critical to each of these tests.

To propose an alternative: can the behavior of the function with regard to the API calls and what not be one set of tests (as most of the things in `expectedCalledStubs` sets up) and another set of tests describing the Ora/spinner behavior created with the success case (which can be set up in a `beforeEach` with stubs, but doesn't have to be asserted because of the first set of tests) and the failure cases (which, after the stubs are set in `beforeEach` can be quickly adjusted in each test case for appropriate scenarios) to make sure the spinner is behaving correctly.

The [docs](https://sinonjs.org/releases/latest/mocks/) for mocks even specify that they shouldn't be used unless they are the system under test as they enforce implementation details. I think this is an important thing to do (the spinner behaves correctly), but these tests are quite large and are conflating the two details about this function in a way that makes it, I think, slightly more difficult to reason about.

Okay, that was long winded. But does that make sense? It will extend the length of this file some, but the tests will be smaller and focus on one detail which should be easier to maintain in the long run.

I should mention - this _would_ mean that the first set of tests, the ones just testing the API calls, would want to have just a [stub](https://sinonjs.org/releases/latest/stubs/) on the spinner (so it doesn't break the output), but the second set of tests _should_ use the mock as done here, as it's the part under test.

Reply:
This is a great advice, thanks! However as @mbleigh mentioned in a previous comment, the spinners shouldn't be placed inside api-call functions and we agreed to move the spinners into the command file, which we do not support unit testing. So an easy option here is just ignoring testing spinner behavior. I think it's possible to think about unit testing the commands, but I would put this as lower priority for now. @nbegley, how about you?

Reply:
+1 to Tri. I agree with Bleigh's idea to move the spinners to the command. Unit testing the commands should be done, but isn't an established pattern yet, so I'd leave that to a separate PR.